include/ginkgo/core/factorization/lu.hpp

```diff
@@ -0,0 +1,147 @@
+/*******************************<GINKGO LICENSE>******************************
+Copyright (c) 2017-2022, the Ginkgo authors
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions
+are met:
+
+1. Redistributions of source code must retain the above copyright
+notice, this list of conditions and the following disclaimer.
+
+2. Redistributions in binary form must reproduce the above copyright
+notice, this list of conditions and the following disclaimer in the
+documentation and/or other materials provided with the distribution.
+
+3. Neither the name of the copyright holder nor the names of its
+contributors may be used to endorse or promote products derived from
+this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
+TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+******************************<GINKGO LICENSE>*******************************/
+
+#include <memory>
+
+
+#include <ginkgo/core/base/composition.hpp>
+#include <ginkgo/core/base/lin_op.hpp>
+#include <ginkgo/core/base/polymorphic_object.hpp>
+#include <ginkgo/core/factorization/factorization.hpp>
+#include <ginkgo/core/matrix/csr.hpp>
+#include <ginkgo/core/matrix/sparsity_csr.hpp>
+
+
+namespace gko {
+namespace experimental {
+namespace factorization {
+
+
+template <typename ValueType, typename IndexType>
+class Lu
+    : public EnablePolymorphicObject<Lu<ValueType, IndexType>, LinOpFactory>,
+      public EnablePolymorphicAssignment<Lu<ValueType, IndexType>> {
+public:
+    struct parameters_type;
+    friend class EnablePolymorphicObject<Lu, LinOpFactory>;
+    friend class enable_parameters_type<parameters_type, Lu>;
+
+    using value_type = ValueType;
+    using index_type = IndexType;
+    using matrix_type = matrix::Csr<value_type, index_type>;
+    using sparsity_pattern_type = matrix::SparsityCsr<value_type, index_type>;
+    using factorization_type = Factorization<value_type, index_type>;
+
+    struct parameters_type
+        : public enable_parameters_type<parameters_type, Lu> {
```

Comment:
I think it can be handled by the usual parameters MACRO like other solvers?

Reply:
For reference: #1162

Reply:
In this case it's not possible, because Lu is a factory-only type without an associated LinOp type.